tests/ModelParser/JMSParserTest.php

```diff
@@ -167,6 +167,42 @@ public function providePropertyType(): iterable
             true,
             'string[]|null',
         ];
+
+        yield [
+            new class() {
+                /**
+                 * @JMS\Type("ArrayCollection<string>")
+                 */
+                private $property;
+            },
+            PropertyTypeArray::class,
+            true,
+            'string[]|\Doctrine\Common\Collections\Collection|null',
```

Comment:
could we make it output `'string[]|\Doctrine\Common\Collections\Collection<string>|null'` ?

Reply:
I adjusted the `__toString()` method via a95fd9aec4fb795ac183a9760b8464324ac4e400. But keep in mind that this is only the representation of the property type. Using something like `Collection<string>` in a docblock would not work, as the `PhpTypeParser` does not support this kind of syntax.